### PR TITLE
Fix find index in BasePermissionsController.js

### LINKED ISSUE #2278: User Devices
The user devices display wrong devices
image

in BasePermissionsController.js line 39 the store find return the wrong index:
image

Also, a primary index is missing on "user_device" table (on fields: userid, deviceid)

### PR DESCRIPTION
**Pull request by Abyss777**
`find` by default uses RegExp
Should fix  #2278

### CODE REVIEW

File: web/app/view/BasePermissionsController.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ Ext.define('Traccar.view.BasePermissionsController', {
                         var i, index;
                         if (success) {
                             for (i = 0; i < records.length; i++) {
-                                index = this.getView().getStore().find('id', records[i].getData().id);
+                                index = this.getView().getStore().findExact('id', records[i].getData().id);
```
Comment (tananaev) [inline web/app/view/BasePermissionsController.js]: Why are we using find here instead of `getById`?

Comment (Abyss777) [inline web/app/view/BasePermissionsController.js]: Frankly, do not now. I used old code:
https://github.com/tananaev/traccar/blob/v3.5/web/app/view/UserDevicesController.js#L37

But sure `getById` will be better

**Comment by Abyss777** [comment]
Used `getById()`
Also replaced all `getData().*` to `getId()` and `get()`
Also used `findExact('type', ` in `NotificationsController.js`

**Comment by tananaev** [comment]
Thanks for fixing the issue and cleaning up the code.

**Pull request merged**